rust/otel-arrow-rust/src/otlp/common.rs

```diff
@@ -369,6 +366,11 @@ impl ProtoBuffer {
         Self { buffer: Vec::new() }
     }
 
+    #[must_use]
+    pub fn to_vec(self) -> Vec<u8> {
```

Comment:
I think we should use the prefix `into` instead of `to` here since the input struct is being consumed.

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Also, we could probably use `into_bytes` as the method name instead of `into_vec`. Similar to the std library method [`String::into_bytes()`](https://doc.rust-lang.org/std/string/struct.String.html#method.into_bytes)

```suggestion
    pub fn into_bytes(self) -> Vec<u8> {
```